src/edu/ucsb/nceas/metacat/dataone/resourcemap/ResourceMapModifier.java

```diff
@@ -0,0 +1,446 @@
+    public final static String TERM_NAMESPACE = DC_TERMS.namespace;
+    public final static String CITO_NAMESPACE = CITO.namespace;
+    public final static String ORE_TER_NAMESPACE = "http://www.openarchives.org/ore/terms/";
+    public final static String RDF_NAMESPACE = "http://www.w3.org/1999/02/22-rdf-syntax-ns#";
+    public final static String AGGREGATION = "#aggregation";
+    public final static String AGENT_URI = "http://purl.org/dc/terms/Agent";
+    
+    private static Log log = LogFactory.getLog(ResourceMapModifier.class);
+    private Identifier oldResourceMapId = null;
+    private Identifier newResourceMapId = null;
+    private Model model = ModelFactory.createDefaultModel();
+    public static String baseURI = null;
+    static {
+        try {
+            String cnUrl = PropertyService.getProperty("D1Client.CN_URL");
+            if(cnUrl.endsWith(SLASH)) {
+                baseURI = cnUrl + RESOLVE;
+            } else {
+                baseURI = cnUrl + SLASH + RESOLVE;
+            }
+        } catch (Exception e) {
+            log.warn("ResourceMapModifier.ResourceMapModifier - couldn't get the value of the property D1Client.CN_URL and Metacat will the default production cn url as the URI base");
+            baseURI = DEFAULT_CN_URI + SLASH + RESOLVE;
+        }
+    }
+    
+    /**
+     * Constructor
+     * @param oldResourceMapId  the identifier of the old resource map which will be modified
+     * @param originalResourceMap  the content of original resource map
+     * @param newResourceMapId  the identifier of the new resource map which will be generated
+     */
+    public ResourceMapModifier(Identifier oldResourceMapId, InputStream originalResourceMap, Identifier newResourceMapId) {
+        this.oldResourceMapId = oldResourceMapId;
+        this.newResourceMapId = newResourceMapId;
+        //read the RDF/XML file
+        model.read(originalResourceMap, null);
+    }
+    
+    /**
+     * Create new resource map by replacing obsoleted ids by new ids.
+     * @param obsoletedBys  a map represents the ids' with the obsoletedBy relationship - the keys are the one need to be obsoleted (replaced); value are the new ones need to be used. They are all science metadata objects
+     * @param newResourceMap  the place where the created new resource map will be written
+     * @param subject  the subject who generates the resource map
+     * @throws UnsupportedEncodingException 
+     */
+    public void replaceObsoletedIds(Map<Identifier, Identifier>obsoletedBys,  OutputStream newResourceMap, Subject subject) throws UnsupportedEncodingException {
+        //replace ids
+        Vector<String> oldURIs = new Vector<String>(); //those uris (resource) shouldn't be aggregated into the new ore since they are obsoleted
+        Vector<String> newURIs = new Vector<String>(); //those uris (resource) should be added into the new aggregation
+        if(obsoletedBys != null) {
```

Comment:
Minor formatting for readability: add spaces in flow control statements - change `if(obsoletedBys != null)` to `if (obsoletedBys != null)`, etc.